tests/xybeamstop.py

```diff
@@ -0,0 +1,164 @@
+import unittest
+import math
+import time
+
+from utils.channel_access import ChannelAccess
+from utils.ioc_launcher import IOCRegister
+
+# Internal Address of device (must be 2 characters)
+ADDRESS = "01"
+
+# MACROS to use for the IOC
+MACROS = {"ADDR": ADDRESS}
+
+PREFIX = "MOT"
+MTR1 = "MOT:MTR0101"
+MTR2 = "MOT:MTR0102"
+
+
+class XybeamstopTests(unittest.TestCase):
+    """
+    Tests for the Larmor X-Y Beamstop
+    """
+    def setUp(self):
+        self._ioc = IOCRegister.get_running("xyBeamstop")
+        self.ca = ChannelAccess()
+        self.ca.wait_for("MOT:ARM:X")
+        self._set_pv_value("ARM:STORE", 0)
+        self._set_x(0.1)
+        self._set_y(0.1)
+
+    def _set_pv_value(self, pv_name, value):
+        self.ca.set_pv_value("{0}:{1}".format(PREFIX, pv_name), value)
+
+    def _set_x(self, value):
+        self._set_pv_value("ARM:X", value)
+
+    def _set_y(self, value):
+        self._set_pv_value("ARM:Y", value)
+
+    def test_GIVEN_x_y_WHEN_read_x_y_THEN_x_y_is_as_expected(self):
```

Comment:
This test name is too generic. You should never use `as_expected` in a test name. The test name should tell you what is expected. It's akin to saying `test_result_correct`.  There seems to be some confusion about the role of GIVEN-WHEN-THEN as well.

GIVEN: The test context. What state is the device in? In this case, there isn't any special state so it can be omitted
WHEN: The action that is taken. The action here isn't reading x and y, it's setting x and y
THEN: The expected result. Here the expected result is that the device has reached its expected position

So something along the lines of:

```
test_WHEN_the_x_y_position_is_set_THEN_the_beam_stop_moves_to_the_supplied_position
```

**Please fix all of the other test names too**